apps/discovery_api/test/integration/discovery_api/data/data_json_test.exs

```diff
@@ -45,22 +49,25 @@ defmodule DiscoveryApi.Data.DataJsonTest do
     end
   end
 
-  defp public_datasets_available?(count) do
-    datasets =
-      "http://localhost:4000/api/v1/data_json"
-      |> get_map_from_url()
-      |> Map.get("dataset", [])
+  test "Returns an additional dataset when we add one via an update", %{organization_id: organization_id} do
+    additional_dataset = TDG.create_dataset(%{technical: %{orgId: organization_id}})
+    Brook.Event.send(DiscoveryApi.instance(), dataset_update(), "integration", additional_dataset)
 
-    if Enum.count(datasets) == count do
-      Enum.all?(datasets, fn dataset -> dataset["accessLevel"] == "public" end)
-    else
-      false
-    end
+    eventually(fn ->
+      assert Enum.count(get_data_json_datasets()) == 3
+    end)
+  end
+
+  defp get_data_json_datasets() do
+    "http://localhost:4000/api/v1/data_json"
```

Comment:
Should this get the port from the config?

Reply:
We probably should, though this is pretty much everywhere at the moment and we're not in a good spot to iterate on our integration tests :( (we have all the prod datasets loaded locally...) - Ben